assembly/std/storage.ts

```diff
@@ -188,49 +282,71 @@ export function appendOf<T>(address: Address, key: T, value: T): void {
 }
 
 /**
- * Checks if the (key, value) exists in the datastore
- * of the callee's address.
+ * Checks if the key-value pair exists in the datastore of the current contract.
+ *
+ * @typeParam T - Can be either `string`, `Args`, or `StaticArray<u8>`.
+ *
+ * @param key - The key to check for existence in the datastore.
+ *
+ * @returns A boolean value indicating whether the key-value pair exists in the datastore or not.
  *
- * @typeParam T - `string`, `Args` or `StaticArray<u8>`
- * @param key -
+ * @throws
+ * - [compilation only] if the `key` type are neither `string`, `StaticArray<u8>`, or `Args`.
  */
 export function has<T>(key: T): bool {
   return env.has(toDatastoreFormat(key));
 }
 
 /**
- * Checks if the (key, value) exists in the datastore
- * of the given address.
+ * Checks if the key-value pair exists in the datastore of the target address.
  *
- * @typeParam T - `string`, `Args` or `StaticArray<u8>`
- * @param address -
- * @param key -
+ * @remarks Can only be called at smart contract generation time by the parent smart contract to the child's address.
  *
+ * @typeParam T - Can be either `string`, `Args`, or `StaticArray<u8>`.
+ *
+ * @param address - The child smart contract `address`.
+ * @param key - The key to check for existence in the datastore.
+ *
+ * @returns A boolean value indicating whether the key-value pair exists in the datastore or not.
+ *
+ * @throws
+ * - if the contract at the given `address` does not exist
+ * - if the caller is not authorized to modify the datastore.
+ * - [compilation only] if the `key` type are neither `string`, `StaticArray<u8>`, or `Args`.
  */
 export function hasOf<T>(address: Address, key: T): bool {
   return env.hasOf(address.toString(), toDatastoreFormat(key));
 }
 
 /**
- *  Sets the executable bytecode of the callee's address.
+ * Sets the executable bytecode of the current address.
  *
- * @privateRemarks
- * TODO: explains failure consequences.
+ * @param bytecode - The bytecode to be set. It should be a `StaticArray<u8>`.
  *
- * @param bytecode -
+ * @throws
+ * - if the caller address does not correspond to a smart contract in the ledger.
+ * - if the bytecode size exceeds maximum allowed size.
```

Comment:
I removed these check because I thought:
- If a smart contract update itself it should have an address. 
- Because it update itself, how could it lack permissions.

Reply:
Am I missing something ?

Reply:
I will add them to reflect the core.

Reply:
You can still call `set_bytecode` from an SC that is not stored on the ledger, in that case the caller address will not correspond to any address in the ledger. And yes you are right about the permissions, if it does not fail for preemptive reasons this error should not happen when its called from a stored SC.

Reply:
🙏